larcorealg/Geometry/AuxDetChannelMapAlg.cxx

```diff
@@ -69,13 +70,14 @@ namespace geo{
 
       HalfCenterWidth = 0.5 * (adsg.HalfWidth1() + adsg.HalfWidth2());
 
-      if(localPoint[2] >= - adsg.Length()/2       &&
-         localPoint[2] <=   adsg.Length()/2       &&
-         localPoint[1] >= - adsg.HalfHeight()     &&
-         localPoint[1] <=   adsg.HalfHeight()     &&
+
+      if(localPoint[2] >= - (adsg.Length()/2 + tolerance)      &&
+         localPoint[2] <=   adsg.Length()/2 + tolerance      &&
+         localPoint[1] >= - (adsg.HalfHeight() + tolerance)    &&
+         localPoint[1] <=   adsg.HalfHeight() + tolerance    &&
          // if AuxDet a is a box, then HalfSmallWidth = HalfWidth
-         localPoint[0] >= - HalfCenterWidth + localPoint[2]*(HalfCenterWidth - adsg.HalfWidth2())/(0.5 * adsg.Length()) &&
-         localPoint[0] <=   HalfCenterWidth - localPoint[2]*(HalfCenterWidth - adsg.HalfWidth2())/(0.5 * adsg.Length())
+         localPoint[0] >= - HalfCenterWidth + localPoint[2]*(HalfCenterWidth - adsg.HalfWidth2())/(0.5 * adsg.Length()) - tolerance &&
+         localPoint[0] <=   HalfCenterWidth - localPoint[2]*(HalfCenterWidth - adsg.HalfWidth2())/(0.5 * adsg.Length()) + tolerance
          ) return a;
```

Comment:
This pattern is repeated enough that it is worth having a data structure that supports this kind of comparison.  I think a clearer coding pattern would be something like:

```c++
namespace lar {
  class BoxBoundaries {
  public:
    Box(double xhw, double y_hw, double zhw, double tolerance = 0.)
      : xhw_{std::abs(xhw)}, yhw_{std::abs(yhw)}, zhw_{std::abs(zhw)}, tolerance_{tolerance}
    {}

    bool contains(double const* point) const
    {
      auto x = point[0];
      auto y = point[1];
      auto z = point[2];
      if (std::abs(x) > xhw_ + tolerance) return false;
      if (std::abs(y) > yhw_ + tolerance) return false;
      if (std::abs(z) > zhw_ + tolerance) return false;
      return true;
    }
  };
}

// ...
BoxBoundaries box{x_halflength, y_halflength, z_halflength, tolerance};
if (box.contains(localPoint)) return a;
```
Of course, you may need to handle a box center which is different than (0,0,0), but I hope this gives you the general idea.  This should result in easier to read code, which is harder to get wrong.

Reply:
Can we make this a feature request for a future release?

Reply:
I'm okay with that.